tensorrt_llm/executor/worker_base.py

```diff
@@ -0,0 +1,834 @@
+import copy
+import datetime
+import enum
+import json
+import weakref
+from pathlib import Path
+from queue import Queue
+from typing import Dict, List, Optional, Tuple, Union
+
+import torch
+
+from tensorrt_llm.logger import logger
+
+from .._utils import (global_mpi_rank, global_mpi_size, mpi_comm, mpi_rank,
+                      nvtx_range_debug)
+from ..bindings import executor as tllm
+from ..builder import ConfigEncoder, Engine, EngineConfig
+from ..llmapi.llm_args import BaseLlmArgs, KvCacheConnectorConfig, PybindMirror
+from ..llmapi.tokenizer import TokenizerBase
+from ..llmapi.tracer import global_tracer
+from ..llmapi.utils import _SyncQueue, print_colored_debug
+from ..lora_helper import LoraConfig
+from ..lora_manager import LoraManager
+from ..metrics import RequestEventTiming
+from ..prompt_adapter_manager import PromptAdapterManager
+from ..runtime import ModelConfig
+from ..runtime.model_runner import _engine_config_to_model_config
+from ..sampling_params import BatchedLogitsProcessor, SamplingParams
+from .executor import GenerationExecutor, IterationResultQueue
+from .ipc import FusedIpcQueue, IpcQueue
+from .postproc_worker import (PostprocParams, PostprocWorker,
+                              PostprocWorkerConfig)
+from .request import GenerationRequest, LoRARequest, PromptAdapterRequest
+from .result import (GenerationResult, LogProbsResult, ResponseWrapper,
+                     compute_logprobs)
+from .utils import (ErrorResponse, IntraProcessQueue, RequestError,
+                    is_llm_response)
+
+__all__ = [
+    "WorkerBase",
+]
+
+
+class WorkerBase(GenerationExecutor):
```

Comment:
I suggest we use `BaseWorker`. This aligns with our established `Base...` prefix convention (e.g., `BaseLlmArgs`, `TorchlmArgs`), which helps keep the codebase consistent and predictable.